### DISCUSSION TITLE
LabD Prefetch : error in data fetching

### DISCUSSION THREAD

**Discussion by fafachen61** [bol-edu/HLS-SOC-Discussions#187] [Q&A]
本組在實作SDRAM controller的prefetch功能的時候，出現了以下情形：

![image](https://github.com/bol-edu/HLS-SOC-Discussions/assets/71396110/a76f4929-3e05-4f27-9809-25df89d2f089)

圖中上方的波型為本組實作的controller，下方為可以正常運作的controller，在這個時間點之前，對比波型兩者的資料獲取與訊號傳遞皆相同，但自addr=000100開始，我們的controller卻會拿到奇怪的值導致程式運行出問題，想請問是否有組別也有遇到類似的情況?

謝謝。

**Comment by ZONG-RUI1216** [comment]
我們組遇到的情況是，要連續發Read的command的話，發完Write command後要再經過4T才能給REFRESH command，否則會他WRITE會出錯(但我還不曉得這原因是甚麼)，所以我後來就把STATE改成IDLE -> WRITE -> WAIT -> IDLE， WAIT delay_ctr設成1就可以了，給你參考。

**Comment by Andy-Ko-0620** [comment]
同學你好，可能是你在寫prefetch功能時沒有同步更新ba_d，導致在addr=000100開始，ontroller拿到奇怪的值導致程式運行出問題。
